### PR TITLE
User can sign in

### PR DESCRIPTION
**Pull request by CristoferMar**
Feature#9 User-can-sign-in

Feature#9 User-can-sign-in-mobile

### CODE REVIEW

File: client/components/list-details.jsx [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
     super(props);
     this.state = {
+      token: null,
       listData: null
     };
   }
 
   componentDidMount() {
+    const { token } = this.context.token;
+    this.setState({ token });
```
Comment (thebearingedge) [inline client/components/list-details.jsx]: You don't have to store the token in state like this, you can refer to `this.context.token` on-demand.

File: client/components/generate-date.jsx [hunk 2]
```diff
@@ -18,9 +20,14 @@ export default class GenerateDate extends React.Component {
   }
 
   componentDidMount() {
+    const { token } = this.context.token;
+    this.setState({ token });
```
Comment (thebearingedge) [inline client/components/generate-date.jsx]: You can use `this.context.token` and remove it from your state.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: client/components/list-details.jsx (hunk 1 above)
Comment (CristoferMar) [reply client/components/list-details.jsx]: Corrected

File: client/components/generate-date.jsx (hunk 2 above)
Comment (CristoferMar) [reply client/components/generate-date.jsx]: Corrected

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**